### PR TITLE
deepin desktop environment: v20 -> v23Beta

### PR DESCRIPTION
**Pull request by wineee**
## Description of changes

deepin V23 Beta2 is officially [released](https://www.deepin.org/en/deepin-v23-beta2-released/) , follow to upgrade DDE

- dtk 5.6.10 -> 5.6.17,   new library [dtkdeclarative](https://github.com/linuxdeepin/dtkdeclarative/#dtkdeclarative), a toolkit based on QtQuick/QtQml
- All Go packages support go mod (buildGoModule) now, buildGoPackage is no longer used
- deepin-music/deepin-image-viewer/deepin-album/dde-launch(pad) refactor using qml (originally qtwidget)

image

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wineee** [comment]
@ofborg build deepin

**Comment by wineee** [comment]
@ofborg build deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

File: pkgs/desktops/deepin/go-package/deepin-desktop-schemas/default.nix [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
     owner = "linuxdeepin";
     repo = pname;
     rev = version;
-    sha256 = "sha256-MboNj0zC3azavDUsmeNNafCcUa0GeoySl610+WOtNww=";
+    hash = "sha256-lvAY0CJ+EohMK16fwIkB7Jq7gTpMhDimPYs9b/IklA4=";
   };
 
-  nativeBuildInputs = [ glib ];
-  buildInputs = [ go-lib ];
+  vendorSha256 = "sha256-q6ugetchJLv2JjZ9+nevUI0ptizh2V+6SByoY/eFJJQ=";
```
Comment (wegank) [inline pkgs/desktops/deepin/go-package/deepin-desktop-schemas/default.nix:19]:
```suggestion
  vendorHash = "sha256-q6ugetchJLv2JjZ9+nevUI0ptizh2V+6SByoY/eFJJQ=";
```

File: pkgs/desktops/deepin/go-package/deepin-pw-check/default.nix [hunk 2]
```diff
@@ -16,20 +12,18 @@
...
   src = fetchFromGitHub {
     owner = "linuxdeepin";
     repo = pname;
     rev = version;
-    sha256 = "sha256-v1Z4ArkrejjOCO1vD+BhfEl9pTfuvKgLM6Ont0IUCQk=";
+    hash = "sha256-kBrkcB0IWGUV4ZrkFzwdPglRgDcnVvYDFhTXS20pKOk=";
   };
 
-  goDeps = ./deps.nix;
+  vendorSha256 = "sha256-L0vUEkUN70Hrx5roIvTfaZBHbbq7mf3WpQJeFAMU5HY=";
```
Comment (wegank) [inline pkgs/desktops/deepin/go-package/deepin-pw-check/default.nix:26]:
```suggestion
  vendorHash = "sha256-L0vUEkUN70Hrx5roIvTfaZBHbbq7mf3WpQJeFAMU5HY=";
```

File: pkgs/desktops/deepin/go-package/dde-daemon/default.nix [hunk 3]
```diff
@@ -32,61 +28,56 @@
...
     (substituteAll {
-      src = ./0004-aviod-use-hardcode-path.patch;
-      inherit dbus;
+      src = ./0002-fix-custom-wallpapers-path.diff;
+      inherit coreutils;
     })
     (substituteAll {
-      src = ./0005-fix-custom-wallpapers-path.diff;
-      inherit coreutils;
+      src = ./0003-aviod-use-hardcode-path.diff;
```
Comment (wegank) [inline pkgs/desktops/deepin/go-package/dde-daemon/default.nix:59]:
```suggestion
      src = ./0003-avoid-use-hardcode-path.diff;
```

Typo.

**Comment by NickCao** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 257400` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
    <li>deepin.deepin-reader</li>
  </ul>
</details>
<details>
  <summary>72 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>


From `deepin.deepin-clone`:
<details>
```
/build/source/app/src/main.cpp: In function 'int main(int, char**)':
/build/source/app/src/main.cpp:100:9: error: 'logger' was not declared in this scope; did you mean 'dlogger'?
  100 |         logger->registerAppender(rollingFileAppender);
      |         ^~~~~~
      |         dlogger
/build/source/app/src/main.cpp:179:5: error: 'logger' was not declared in this scope; did you mean 'dlogger'?
  179 |     logger->registerCategoryAppender("deepin.ghost", consoleAppender);
      |     ^~~~~~
      |     dlogger
```
</details>

From `deepin.deepin-reader`:

<details>
```
sidebar/BookMarkWidget.cpp: In member function 'void BookMarkWidget::initWidget()':
sidebar/BookMarkWidget.cpp:36:13: error: 'DGuiApplicationHelper' has not been declared
   36 |     connect(DGuiApplicationHelper::instance(), &DGuiApplicationHelper::themeTypeChanged, this, &BookMarkWidget::onUpdateTheme);
      |             ^~~~~~~~~~~~~~~~~~~~~
sidebar/BookMarkWidget.cpp:36:49: error: 'DGuiApplicationHelper' has not been declared
   36 |     connect(DGuiApplicationHelper::instance(), &DGuiApplicationHelper::themeTypeChanged, this, &BookMarkWidget::onUpdateTheme);
      |                                                 ^~~~~~~~~~~~~~~~~~~~~
sidebar/BookMarkWidget.cpp: In member function 'void BookMarkWidget::onUpdateTheme()':
sidebar/BookMarkWidget.cpp:193:5: error: 'DPalette' was not declared in this scope; did you mean 'Dtk::Gui::DPalette'?
  193 |     DPalette plt = Dtk::Gui::DGuiApplicationHelper::instance()->applicationPalette();
      |     ^~~~~~~~
      |     Dtk::Gui::DPalette
In file included from /nix/store/mq52m0jfc9abqhzlpx8i91ml30cg7yd7-dtkgui-5.6.17-dev/include/dtk5/DGui/DPalette:1,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/dstyleditemdelegate.h:10,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/DStyledItemDelegate:1,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/dlistview.h:11,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/DListView:1,
                 from sidebar/SideBarImageListview.h:9,
                 from sidebar/BookMarkWidget.cpp:8:
/nix/store/mq52m0jfc9abqhzlpx8i91ml30cg7yd7-dtkgui-5.6.17-dev/include/dtk5/DGui/dpalette.h:16:7: note: 'Dtk::Gui::DPalette' declared here
   16 | class DPalette : public QPalette
      |       ^~~~~~~~
sidebar/BookMarkWidget.cpp:194:5: error: 'plt' was not declared in this scope
  194 |     plt.setColor(Dtk::Gui::DPalette::Background, plt.color(Dtk::Gui::DPalette::Base));
      |     ^~~
```
</details>

File: pkgs/desktops/deepin/core/dde-application-manager/default.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  version = "1.0.19";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    hash = "sha256-1P265xqlL/wML66nKdfTgkRx6MCpLwrt5rXu+CyeShU=";
+  };
+
+  # remove this patch after next release
```
[the remark this answers is not in the document]
Comment (wineee) [reply pkgs/desktops/deepin/core/dde-application-manager/default.nix:28]: dde-application-manager use `develop/refact` branch. The new application-manager does not have a hard coded path, but the interface has changed and needs to wait for other dde components to adapt

**Comment by wineee** [comment]
deepin-reader: fix build in 6.0.5

deepin-clone: mark as broken， Lack of maintenance in upstream, this package may be removed in the future

**Comment by pbsds** [comment]
Error in `deepin.dde-gsettings-schemas`:

```
cp: missing destination file operand after '/nix/store/7zrawcajf0d58jamclfamzwn1f7mqpza-nixos-gsettings-desktop-schemas/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas'
Try 'cp --help' for more information.
```

**Comment by wineee** [comment]
@ofborg build deepin.dde-gsettings-schemas

**Comment by wineee** [comment]
@ofborg build deepin

**Comment by wineee** [comment]
@ofborg build pantheon.elementary-gsettings-schemas

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by pbsds** [comment]
Played around in a VM, everything i tried worked except the video player, but MPV had the same issue so it is likely the vm setup that did it.

<details>

Screenshot from 2023-10-28 02-48-18
Screenshot from 2023-10-28 02-48-40
Screenshot from 2023-10-28 02-48-57
Screenshot from 2023-10-28 02-49-13
Screenshot from 2023-10-28 02-49-56
Screenshot from 2023-10-28 02-50-23
Screenshot from 2023-10-28 02-50-52
Screenshot from 2023-10-28 02-51-30
Screenshot from 2023-10-28 02-55-02
Screenshot from 2023-10-28 02-56-46
Screenshot from 2023-10-28 02-56-53
Screenshot from 2023-10-28 02-59-20
Screenshot from 2023-10-28 03-00-25
Screenshot from 2023-10-28 03-00-40
Screenshot from 2023-10-28 03-00-43
Screenshot from 2023-10-28 03-01-46
Screenshot from 2023-10-28 03-02-06

</detail>

**Review by pbsds — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-session</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>
<details>
  <summary>66 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
  </ul>
</details>

**Comment by wineee** [comment]
> Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package marked as broken and skipped:
> 1 package blacklisted:
> 6 packages failed to build:
> 66 packages built:

I don't have an arm linux device, but [nixosTests.deepin on aarch64 Linux](https://logs.ofborg.org/?key=nixos/nixpkgs.257400&attempt_id=db823c89-fdcf-4a62-a7fd-f714c31c7362) success, could you please add more error log

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-session</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>
<details>
  <summary>66 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
  </ul>
</details>

**Comment by wegank** [comment]
`libsecret` fails on my machine, possibly unrelated.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>72 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Comment by pbsds** [comment]
Do you intend to push any more changes to this?

**Comment by wineee** [comment]
> Do you intend to push any more changes to this?


I have rebased for https://github.com/NixOS/nixpkgs/pull/264280 , no other changes

**Comment by wineee** [comment]
rebased to reslove merge conflict

**Comment by K900** [comment]
We don't want to ship a beta version of DDE in 23.11, right?

**Comment by wineee** [comment]
> We don't want to ship a beta version of DDE in 23.11, right?

right, It's better to ship it in 24.05-unstable, allow enough time for testing and improvement
(DDE has updated many components after the release of v23Beta. the beta version more refers to deepin os, and DDE has not yet an independent version.)

**Comment by wineee** [comment]
After NixOS-23.11 branch-off,   this pr is ready for review now

image

**Pull request merged**